libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -762,106 +762,109 @@ void YamlConfiguration::parsePortYaml(YAML::Node *portNode, core::ProcessGroup *
+        std::shared_ptr<core::Connectable> proc = std::dynamic_pointer_cast<core::Connectable>(processor);
+        if (proc != 0) {
+          logger_->log_warn("Received property %s with value %s but is not one of the properties for %s. Attempting to add as dynamic property.", propertyName, rawValueString, proc->getName());
+          if (!processor->setDynamicProperty(propertyName, rawValueString)) {
+            logger_->log_warn("Unable to set the dynamic property %s with value %s", propertyName.c_str(), rawValueString.c_str());
           } else {
-            auto typedValue = propertyValueNode.as<std::string>();
-            coercedValue = typedValue;
-          }
-        } catch (...) {
-          std::string eof;
-          bool exit_on_failure = false;
-          if (configuration_->get(Configure::nifi_flow_configuration_file_exit_failure, eof)) {
-            utils::StringUtils::StringToBool(eof, exit_on_failure);
-          }
-          logger_->log_error("Invalid conversion for field %s. Value %s", myProp.getName(), propertyValueNode.as<std::string>());
-          if (exit_on_failure) {
-            std::cerr << "Invalid conversion for " << myProp.getName() << " to " << defaultType.name() << std::endl;
-          } else {
-            coercedValue = defaultValue;
+            logger_->log_warn("Dynamic property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
           }
         }
-        std::string rawValueString = propertyValueNode.as<std::string>();
-        if (!processor->setProperty(myProp, coercedValue)) {
-          std::shared_ptr<core::Connectable> proc = std::dynamic_pointer_cast<core::Connectable>(processor);
-          if (proc != 0) {
-            logger_->log_warn("Received property %s with value %s but is not one of the properties for %s. "
-                              "Attempting to add as dynamic property.",
-                              propertyName, rawValueString, proc->getName());
-            if (!processor->setDynamicProperty(propertyName, rawValueString)) {
-              logger_->log_warn("Unable to set the dynamic property %s with value %s", propertyName.c_str(), rawValueString.c_str());
-            } else {
-              logger_->log_warn("Dynamic property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
-            }
-          }
-        } else {
-          logger_->log_debug("Property %s with value %s set", propertyName.c_str(), rawValueString.c_str());
-        }
       }
     }
   }
+}
+
+PropertyValue YamlConfiguration::getValidatedProcessorPropertyForDefaultTypeInfo(const core::Property& propertyFromProcessor, const YAML::Node& propertyValueNode) {
+  PropertyValue defaultValue;
+  defaultValue = propertyFromProcessor.getDefaultValue();
+  auto defaultType = defaultValue.getTypeInfo();
+  try {
+    PropertyValue coercedValue = defaultValue;
+    if (defaultType == typeid(int64_t)) {
+      coercedValue = propertyValueNode.as<int64_t>();
```

Comment:
Hm... this has changed quite a bit.  For example, why don't we need the case for `uint64_t` any longer?

Also, do we have enough unit tests to make sure the new code works in the same way as the old code?

Reply:
Honestly, I do not quite remember why we do not need `uint64_t` anymore :S

It probably has something to do with the uint64 nodes being parsed as string even though their value type matches here:
```c++
class DataSizeValue : public TransformableValue, public state::response::UInt64Value {
```

There has been a discussion about it where @szaszm replied that he agreed on the removal, but gah this is old. This should be though thoroughly tested as there are many integration tests that directly rely on yamlparsing and even some yamlconfiguration tests as well.

Reply:
I don't remember the discussion, so some pointers to context would be appreciated. :slightly_smiling_face: